keras/callbacks.py

```diff
@@ -694,13 +703,16 @@ def on_epoch_end(self, epoch, logs=None):
             if epoch % self.histogram_freq == 0:
                 # TODO: implement batched calls to sess.run
                 # (current call will likely go OOM on GPU)
+
+                val_data = self.validation_data
+                tensors = self.model.inputs + \
+                    self.model.targets + \
+                    self.model.sample_weights
+
                 if self.model.uses_learning_phase:
-                    cut_v_data = len(self.model.inputs)
-                    val_data = self.validation_data[:cut_v_data] + [0]
-                    tensors = self.model.inputs + [K.learning_phase()]
-                else:
-                    val_data = self.validation_data
-                    tensors = self.model.inputs
+                    tensors += [K.learning_phase()]
+
+                assert len(val_data) == len(tensors)
```

Comment:
Is this `assert` ever not verified (e.g. due to a user error)? If so, then you should not use an assert but instead throw a specific Exception subclass, with a helpful and actionable error message.

If the assert is just for debugging / sanity-checking, then it's fine.